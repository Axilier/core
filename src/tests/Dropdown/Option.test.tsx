import React from 'react';
import { mount } from 'enzyme';
import { Option } from '../../index';

const onClick = jest.fn();
const option = mount(<Option onClick={onClick}>test</Option>);

describe('when the option is used in an application', () => {
    it('should render the contained option correctly', () => {
        expect(option).toMatchSnapshot();
    });
    describe('when `onClick` is defined, and the option is pressed', () => {
        beforeAll(() => option.simulate('click'));
        it('should run the onClick function', () => {
            expect(onClick).toHaveBeenCalled();
        });
    });
    describe('when `onClick` is not defined, and the option is pressed', () => {
        beforeAll(() => {
            option.setProps({ onClick: null });
            option.simulate('click');
        });
        it('should do nothing', () => {
            expect(true).toBeTruthy();
        });
    });
});
