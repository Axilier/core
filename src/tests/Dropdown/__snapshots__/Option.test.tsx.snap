// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the option is used in an application should render the contained option correctly 1`] = `
<Option
  onClick={[MockFunction]}
>
  <div
    className="option"
    onClick={[Function]}
    role="button"
  >
    test
  </div>
</Option>
`;
